data/scripts/dependency-resolver.py

```diff
@@ -108,31 +141,59 @@ def compile_test(source, compiler, cflags, ldflags):
 
     return status
 
-def handle_ccode_check(args, conf):
+def set_makefile_compflags(flags, prefix, suffix):
+    append_to = flags.get("appent_to")
+    flag_value = flags.get("value")
+
+    if not flag_value:
+        return
+
+    if append_to:
+        context.add_append_makefile_var(append_to, flag_value)
+    else:
+        context.add_cond_makefile_var("%s_%s" % (prefix, suffix),
+                                      flag_value)
+
+def handle_ccode_check(args, conf, context):
     dep = conf["dependency"].upper()
     source = ""
-    for i in conf["headers"]:
-        source += "#include %s\n" % i
 
+    cflags = conf.get("cflags")
+    ldflags = conf.get("ldflags")
+
+    defines = conf.get("defines")
```

Comment:
You can simplify this get + if with:

    for define in conf.get("defines", []):
        source += ...